From 6c57ca52752701e7222f09836393e766a359a36b Mon Sep 17 00:00:00 2001
From: Enrico Berti <hello@enricoberti.com>
Date: Thu, 9 Oct 2014 23:41:03 +0200
Subject: [PATCH 0019/1173] HUE-2371 [sentry] Sentry URI should be created
 only with a ALL permission

Set select2 to All and disabled it
---
 apps/security/src/security/templates/hive.mako |    2 +-
 apps/security/static/js/common.ko.js           |    6 ++++++
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/apps/security/src/security/templates/hive.mako b/apps/security/src/security/templates/hive.mako
index 6c926fd..f9622e4 100644
--- a/apps/security/src/security/templates/hive.mako
+++ b/apps/security/src/security/templates/hive.mako
@@ -67,7 +67,7 @@ ${ layout.menubar(section='hive') }
     </div>
     <input type="text" data-bind="filechooser: $data.URI, enable: privilegeType() == 'uri'" placeholder="URI">
 
-    <select data-bind="options: $root.availableActions, select2: { update: $data.action, type: 'action'}" style="width: 100px"></select>
+    <select data-bind="options: $root.availableActions, select2: { update: $data.action, type: 'action', readonly: (privilegeType() == 'uri'), readonlySetTo: function(){ (privilegeType() == 'uri') ? $data.action($root.availableActions()[0]) : $data.action('ALL') }}" style="width: 100px"></select>
 
     <div class="new-line-if-small">
       <label class="checkbox"><input type="checkbox" data-bind="checked: grantOption"> ${ _('With grant') }</label>
diff --git a/apps/security/static/js/common.ko.js b/apps/security/static/js/common.ko.js
index a5275b2..05e6ddb 100644
--- a/apps/security/static/js/common.ko.js
+++ b/apps/security/static/js/common.ko.js
@@ -100,6 +100,12 @@ ko.bindingHandlers.select2 = {
     if (typeof valueAccessor().update != "undefined") {
       $(element).select2("val", valueAccessor().update());
     }
+    if (typeof valueAccessor().readonly != "undefined") {
+      $(element).select2("readonly", valueAccessor().readonly);
+      if (typeof valueAccessor().readonlySetTo != "undefined") {
+        valueAccessor().readonlySetTo();
+      }
+    }
   }
 };
 
-- 
1.7.9.5

